src/crypto/equihash.cpp

```diff
@@ -268,11 +268,24 @@ std::set<std::vector<eh_index>> Equihash<N,K>::BasicSolve(const eh_HashState& ba
         LogPrint("pow", "- Sorting list\n");
         std::sort(X.begin(), X.end(), CompareSR(hashLen));
         LogPrint("pow", "- Finding collisions\n");
-        for (int i = 0; i < X.size() - 1; i++) {
-            FullStepRow<FinalFullWidth> res(X[i], X[i+1], hashLen, lenIndices, 0);
-            if (res.IsZero(hashLen) && DistinctIndices(X[i], X[i+1], hashLen, lenIndices)) {
-                solns.insert(res.GetIndices(hashLen, 2*lenIndices));
+        int i = 0;
+        while (i < X.size() - 1) {
+            int j = 1;
+            while (i+j < X.size() &&
```

Comment:
Style, should be `i + j` (this is the best I got because I don't understand Equihash well enough!)